library/debug.c

```diff
@@ -227,6 +227,8 @@ void mbedtls_debug_print_mpi( const mbedtls_ssl_context *ssl, int level,
         NULL == ssl->conf        ||
         NULL == ssl->conf->f_dbg ||
         NULL == X                ||
+        NULL == X->p             ||
```

Comment:
This is not needed: `X->p` can only be null when `X->n == 0`.

It can make sense for a debug function to support malformed objects, but in this case it should print a message like “malformed mpi” and not just silently do nothing.